src/controllers/commands/CommandController.cpp

```diff
@@ -1355,6 +1355,126 @@ void CommandController::initialize(Settings &, Paths &paths)
             (void)words;  // unused
             return deleteMessages(channel, QString());
         });
+
+    this->registerCommand("/mod", [](const QStringList &words, auto channel) {
+        if (words.size() < 2)
+        {
+            channel->addMessage(makeSystemMessage(
+                "Usage: \"/mod <username>\" - Grant moderator status to a "
+                "user. Use \"/mods\" to list the moderators of this channel."));
+            return "";
+        }
+
+        auto currentUser = getApp()->accounts->twitch.getCurrent();
+        if (currentUser->isAnon())
+        {
+            channel->addMessage(
+                makeSystemMessage("You must be logged in to mod someone!"));
+            return "";
+        }
+
+        auto *twitchChannel = dynamic_cast<TwitchChannel *>(channel.get());
+        if (twitchChannel == nullptr)
+        {
+            channel->addMessage(makeSystemMessage(
+                "The /mod command only works in Twitch channels"));
+            return "";
+        }
+
+        if (twitchChannel->roomId() != currentUser->getUserId())
+        {
+            channel->addMessage(makeSystemMessage(
+                "You may only mod people in your own channel."));
+            return "";
+        }
+
+        auto target = words.at(1);
+        stripChannelName(target);
+
+        getHelix()->getUserByName(
+            target,
+            [currentUser, channel](const HelixUser &targetUser) {
+                getHelix()->addChannelModerator(
+                    currentUser->getUserId(), targetUser.id,
```

Comment:
Let's avoid doing client-side checks where possible

This means removing the early out, handling the 401 error correctly and passing `twitchChannel->roomId()` to the `addChannelModerator` function instead of `currentUser->getUserId()`

Reply:
removed, added handling for that particular error